src/main/java/org/opentripplanner/routing/algorithm/mapping/GraphPathToItineraryMapper.java

```diff
@@ -211,6 +211,7 @@ private static CoordinateArrayListSequence makeCoordinates(Edge[] edges) {
             var modeChange = backMode != forwardMode && backMode != null && forwardMode != null;
             var parkingChange = backState.isVehicleParked() != forwardState.isVehicleParked();
             var rentalChange = isRentalPickUp(backState) || isRentalDropOff(backState);
+            var flexEdge = forwardState.getBackEdge() instanceof FlexTripEdge;
```

Comment:
The problem here is that the flex `BUS` state consists of `null, BUS, null` modes which the above `modeChange` variable doesn't find.

Reply:
I've updated the logic to be clearer.